lib/gui/app.js

```diff
@@ -83,23 +84,44 @@ app.run(() => {
 app.run((AnalyticsService, ErrorService, UpdateNotifierService, SelectionStateModel) => {
   AnalyticsService.logEvent('Application start');
 
-  if (UpdateNotifierService.shouldCheckForUpdates() && !process.env.ETCHER_DISABLE_UPDATES) {
-    AnalyticsService.logEvent('Checking for updates');
+  const shouldCheckForUpdates = UpdateNotifierService.shouldCheckForUpdates();
 
-    UpdateNotifierService.isLatestVersion().then((isLatestVersion) => {
+  if (!shouldCheckForUpdates || process.env.ETCHER_DISABLE_UPDATES) {
+    AnalyticsService.logEvent('Not checking for updates', {
+      shouldCheckForUpdates,
+      disableUpdatesEnvironmentVariable: process.env.ETCHER_DISABLE_UPDATES
+    });
+    return;
+  }
 
-      // In case the internet connection is not good and checking the
-      // latest published version takes too long, only show notify
-      // the user about the new version if he didn't start the flash
-      // process (e.g: selected an image), otherwise such interruption
-      // might be annoying.
-      if (!isLatestVersion && !SelectionStateModel.hasImage()) {
+  AnalyticsService.logEvent('Checking for updates', {
+    currentVersion: packageJSON.version
+  });
 
-        AnalyticsService.logEvent('Notifying update');
```

Comment:
I initially moved it in `UpdateNotifierService.notify()`, then removed it from there so all updates related logging is in `app.js`, but never added it back... It would get logged as an open modal (`Open modal ({"Template":"./components/update-notifier/templates/update-notifier-modal.tpl.html"})`), but an explicit log might be useful as well. I'll add it back.